### PR TITLE
Issue 153

### PR DESCRIPTION
**Pull request by fassko**
More or less ready, will add something until tomorrow noon

### CODE REVIEW

File: _drafts/2020-02-13-issue-153.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+title: ! 'Issue #153'
+author: fassko
+sponsor:
+    link: TODO
+    heading: TODO
+    body: TODO
+    displaylink: TODO
+---
+
+The last two weeks were super busy with Apple. Starting with new [Xcode 11.4 beta](https://developer.apple.com/documentation/xcode_release_notes/xcode_11_4_beta_release_notes), [Swift Crypto](https://github.com/apple/swift-crypto) library releases and making [Swift Playgrounds](https://apps.apple.com/us/app/swift-playgrounds/id1496833156) available for Mac.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > ~~Starting with new...~~ A new [Xcode 11.4 beta](https://developer.apple.com/documentation/xcode_release_notes/xcode_11_4_beta_release_notes) was released, a new [Swift Crypto](https://github.com/apple/swift-crypto) library was released and [Swift Playgrounds](https://apps.apple.com/us/app/swift-playgrounds/id1496833156) is now available for Mac.

File: _drafts/2020-02-13-issue-153.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+On top of that Swift repository has now more than 100,000 commits!
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > On top of that **the** Swift repository has now more than 100,000 commits!

I would link the tweet here and remove it from below

File: _drafts/2020-02-13-issue-153.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+### Starter tasks
+
+> TODO
+
+### Podcasts
+
+In [episode 84 of Swift Unwrapped](https://spec.fm/podcasts/swift-unwrapped/316012), [Jesse](https://twitter.com/jesse_squires) and [JP](https://twitter.com/simjp) discuss Swift in 2020.
+
+In latest [episode 7 or Swift Community Podcast](https://www.swiftcommunitypodcast.org/episodes/7) [Tim Condon](https://twitter.com/0xTim) talks to [Kaitlin Mahar](https://twitter.com/k__mahar), [Siemen Sikkema](https://twitter.com/siemensikkema), [Tanner Nelson](https://twitter.com/tanner0101) and [Ian Partridge](https://twitter.com/alfa) about the current state of Swift on the server, and what the future might have in store
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > In ~~latest~~ episode 7 ~~or~~ **of**

File: _drafts/2020-02-13-issue-153.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+### News and community
+
+Xcode 11.4 betas are out with Swift 5.2, which has a ton of compiler performance and quality of life improvements, including the new diagnostic engine, [https://twitter.com/jckarter/status/1225148705814106112](tweeted) [Joe Groff](https://twitter.com/jckarter).
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > Xcode 11.4 beta is out with Swift 5.2, which has a ton of compiler performance and quality of life improvements, including the [new diagnostic engine](https://twitter.com/jckarter/status/1225148705814106112).

File: _drafts/2020-02-13-issue-153.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+[Cory Benfield](https://twitter.com/Lukasaoz) released [Swift Crypto](https://github.com/apple/swift-crypto) on the [dotSwift](https://www.dotswift.io/) conference [stage](https://twitter.com/NicoonGuitar/status/1224376366092767233). He wrote an [article](https://swift.org/blog/crypto/) on Swift.org website. As well some folks started a [discussion](https://forums.swift.org/t/native-implementations-and-boringssl-backed-apple-platform-deployments/33404).
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > ~~on~~ **at** the dotSwift...

> on Swift.org website, ~~as well some~~ **and some folks**...

File: _drafts/2020-02-13-issue-153.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+> Swift Crypto is a new Swift package that brings the fantastic APIs of [Apple CryptoKit](https://developer.apple.com/documentation/cryptokit) to the wider Swift community. This will allow Swift developers, regardless of the platform on which they deploy their applications, to access these APIs for a common set of cryptographic operations.
+
+[Slava Pestov](https://twitter.com/slava_pestov) [tweeted](https://twitter.com/slava_pestov/status/1227438709416628224) that the Swift repository now has over 100,000 commits! That’s a lot of commits — from a lot of different people. The most prolific contributor is
+[Doug Gregor](https://twitter.com/dgregor79) but he’s only responsible for 6% of the total.
+
+Apple released [Swift Playgrounds for Mac](https://apps.apple.com/us/app/swift-playgrounds/id1496833156) - makes it fun to learn and experiment with code. Previously it was available only for iPad.
+
+The release of Swift 5.1.4 for Linux has been [announced](https://forums.swift.org/t/announcing-swift-5-1-4-for-linux/33443). New downloads are available on [swift.org](https://swift.org/download/#swift-514). Official Docker images are also available.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > The release of Swift 5.1.4 for Linux ~~has been~~ **was**...

File: _drafts/2020-02-13-issue-153.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+[Ole Begemann](https://twitter.com/olebegemann) wrote two great articles. [First one](https://oleb.net/2020/swift-test-discovery/) is about automatic test discovery in Swift on Linux. In [second one](https://oleb.net/2020/swift-docker-linux/) he talks about how to test Swift packages on Linux using Docker.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > One about [automatic test discovery in Swift on Linux](https://oleb.net/2020/swift-test-discovery/) and another on [how to test Swift packages on Linux using Docker](https://oleb.net/2020/swift-docker-linux/).

File: _drafts/2020-02-13-issue-153.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+[John Sundell](https://twitter.com/johnsundell) wrote an [article](https://www.swiftbysundell.com/articles/slicing-swift-collections/) explaining slicing Swift collections.
+
+[Antoine van der Lee](https://twitter.com/twannl) explaining in his [blog post](https://www.avanderlee.com/swift/creating-swift-package-manager-framework/) how to create the Swift Package framework in Xcode.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > ~~explaining~~ **explained**

File: _drafts/2020-02-13-issue-153.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+[Point-Free](https://twitter.com/pointfreeco) folks [announced](https://twitter.com/pointfreeco/status/1224742851537514497) a library called [CasePaths](https://github.com/pointfreeco/swift-case-paths) that brings the power and ergonomics of key paths to enums.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > **The** Point-Free folks ...

File: _drafts/2020-02-13-issue-153.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+[Alejandro](https://twitter.com/aalonso128) [tweeted](https://twitter.com/aalonso128/status/1224694162055999493) that Swift nightly builds are now available on [swift.godbolt.org](https://swift.godbolt.org/). Gobolt is a compiler explorer tool.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > [Alejandro tweeted](https://twitter.com/aalonso128/status/1224694162055999493) that ...

File: _drafts/2020-02-13-issue-153.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+[Paul Hudson](https://twitter.com/twostraws) [tweeted](https://twitter.com/twostraws/status/1225207234185121793) that [whatsnewinswift.com](https://www.whatsnewinswift.com/?from=5.1&to=5.2) now covers Swift 5.2 and other changes between the versions.
+Along with that, he released a tool [Sitrep](https://github.com/twostraws/Sitrep) that is a source code analyzer for Swift projects, giving you a high-level overview of your code.
+
+Two awesome blog posts by [Mattt](https://twitter.com/mattt) about [Language Server Protocol](https://nshipster.com/language-server-protocol/) and [Swift Development with Visual Studio Code](https://nshipster.com/vscode/).
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > ~~Two awesome blog posts by [Mattt](https://twitter.com/mattt)~~ **[Mattt](https://twitter.com/mattt) wrote two awesome blog posts** ....

File: _drafts/2020-02-13-issue-153.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+[Antoine van der Lee](https://twitter.com/twannl) explaining in his [blog post](https://www.avanderlee.com/swift/creating-swift-package-manager-framework/) how to create the Swift Package framework in Xcode.
+
+[Point-Free](https://twitter.com/pointfreeco) folks [announced](https://twitter.com/pointfreeco/status/1224742851537514497) a library called [CasePaths](https://github.com/pointfreeco/swift-case-paths) that brings the power and ergonomics of key paths to enums.
+
+[Alejandro](https://twitter.com/aalonso128) [tweeted](https://twitter.com/aalonso128/status/1224694162055999493) that Swift nightly builds are now available on [swift.godbolt.org](https://swift.godbolt.org/). Gobolt is a compiler explorer tool.
+
+[Paul Hudson](https://twitter.com/twostraws) [tweeted](https://twitter.com/twostraws/status/1225207234185121793) that [whatsnewinswift.com](https://www.whatsnewinswift.com/?from=5.1&to=5.2) now covers Swift 5.2 and other changes between the versions.
+Along with that, he released a tool [Sitrep](https://github.com/twostraws/Sitrep) that is a source code analyzer for Swift projects, giving you a high-level overview of your code.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > released a tool **called** ...

File: _drafts/2020-02-13-issue-153.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+[Doug Gregor](https://twitter.com/dgregor79) merged [a pull request](https://github.com/apple/swift/pull/29728) that makes it easy to extend Swift function builders to support local `let`'s.
+
+> Sink the constraint generation and solution application of initialization patterns, including all
+of the logic for property wrappers, from the high-level entry point
+`typeCheckBinding` down into the lower-level handling for
+solution application targets.
+
+### Accepted proposals
+
+[SE-0270](https://github.com/apple/swift-evolution/blob/master/proposals/0270-rangeset-and-collection-operations.md): *Add Collection Operations on Noncontiguous Elements* was [accepted](https://forums.swift.org/t/accepted-se-0270-add-collection-operations-on-noncontiguous-elements/33270). Although some [discussions](https://forums.swift.org/t/evolution-process-discussion/33272) started about that.
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > ~~Although some [discussions](https://forums.swift.org/t/evolution-process-discussion/33272) started about that.~~ **See the [discussions here](https://forums.swift.org/t/evolution-process-discussion/33272).**

File: _drafts/2020-02-13-issue-153.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
+> In short, the original pitch was to create a special-case, compiler-level mechanism to generate Equatable conformance for all tuples whose elements themselves conform to Equatable. The conformance would be automatic, without any means to opt-in or opt-out.
+> 
+> There exists a fairly wide-spread and long-standing desire to enable protocol conformance for tuples and perhaps other structural types, and especially so with respect to ubiquitous protocols, like Equatable, Comparable, Hashable and Codable. Presently, we are not in a position to facilitate custom protocol conformance as a general feature for tuples or other structural types. It is anticipated that that sort of feature will be created, eventually.
+
+[Taneer Nelson](https://twitter.com/tanner0101) [shared](https://forums.swift.org/t/january-23rd-2020/33715) Swift on Server WorkGroup January 23rd, 2020 meeting notes.
+
+### Finally
+
+Should Swift library [include](https://twitter.com/jesse_squires/status/1224456344792530949) a lot of Swift code? 🤔
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > Should **a** Swift ...

File: _drafts/2020-02-13-issue-153.md [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+
+Small [change](https://github.com/apple/swift/commit/77da6bd3a7c3c2ec02bb0f13e7341cf867cab0d3), but big [impact](https://twitter.com/dgregor79/status/1224537246998487040). 🙃
```
Comment (jessesquires) [inline _drafts/2020-02-13-issue-153.md]: > **A** small...

**Comment by BasThomas** [comment]
Thanks for the initial review @jessesquires 😊

Looking forward to giving it a read soon too :)

**Comment by fassko** [comment]
Thanks @jessesquires ! :) 

I hope I didn't miss anything.

**Comment by fassko** [comment]
@BasThomas please check, I made small correction then I can merge :)

**Comment by BasThomas** [comment]
@fassko this is ready to merge 😊

**Pull request merged**